src/Surfnet/StepupGateway/GatewayBundle/Controller/SecondFactorController.php

```diff
@@ -96,72 +92,65 @@ public function selectSecondFactorForVerificationAction($authenticationMode, Req
         } catch (LoaCannotBeGivenException $e) {
             // Log the message of the domain exception, this contains a meaningful message.
             $logger->notice($e->getMessage());
-
             return $this->forward(
                 'SurfnetStepupGatewayGatewayBundle:Gateway:sendLoaCannotBeGiven',
                 ['authenticationMode' => $authenticationMode]
             );
         }
 
         $logger->notice(sprintf('Determined that the required Loa is "%s"', $requiredLoa));
-
         if ($this->getStepupService()->isIntrinsicLoa($requiredLoa)) {
             $this->get('gateway.authentication_logger')->logIntrinsicLoaAuthentication($originalRequestId);
             return $this->forward($context->getResponseAction());
         }
 
+        // The preconditions must be met in order to give SSO on 2FA
+        // 1: AuthNRequest is not force authn. 2: The SP allows SSO on 2FA.
+        if ($this->getCookieService()->preconditionsAreMet($context)) {
+            // Now read the SSO cookie
+            $ssoCookie = $this->getCookieService()->read($request);
+            // Test if the SSO cookie can satisfy the second factor authentication requirements
+            if ($this->getCookieService()->maySkipAuthentication($requiredLoa->getLevel(), $identityNameId, $ssoCookie)) {
+                $logger->notice('Skipping second factor authentication. Required LoA was met by the LoA recorded in the cookie');
```

Comment:
Could you add the required LoA and the LoA found in the cookie to the log message?

Reply:
I most certainly can

Reply:
In retrospect, I think its better to remove this notice, as the maySkip method itself already provided that more detailed message you linked in your comment.